app/dao/annual_billing_dao.py

```diff
@@ -0,0 +1,47 @@
+from app import db, create_uuid
+from app.dao.dao_utils import (
+    transactional,
+    version_class
+)
+from app.models import AnnualBilling
+from datetime import datetime
+from app.service.utils import get_current_financial_year_start_year
+
+
+def dao_get_annual_billing(service_id):
+    return AnnualBilling.query.filter_by(
+        service_id=service_id,
+    ).order_by(AnnualBilling.financial_year_start).all()
+
+
+def dao_create_or_update_annual_billing_for_year(annual_billing):
+    db.session.add(annual_billing)
+    db.session.commit()
+
+
+def dao_get_free_sms_fragment_limit_for_year(service_id, year):
+
+    return AnnualBilling.query.filter_by(
+        service_id=service_id,
+        financial_year_start=year
+    ).first()
+
+
+def dao_get_all_free_sms_fragment_limit(service_id):
+
+    return AnnualBilling.query.filter_by(
+        service_id=service_id,
+    ).order_by(AnnualBilling.financial_year_start).all()
+
+
+def insert_annual_billing(service):
```

Comment:
should this be dao_insert....?
I thought the convention was that public methods started with dao?

Reply:
Changed